src/java/org/jivesoftware/openfire/roster/RosterManager.java

```diff
@@ -652,23 +662,24 @@ private void groupUserAdded(Group group, Collection<JID> users, JID addedUser) {
                 // Get the roster to update
                 Roster roster = null;
                 if (server.isLocal(userToUpdate)) {
-                    // Check that the user exists, if not then continue with the next user
+                    // Get the roster. If the user does not exist then continue with the next user
                     try {
-                        UserManager.getInstance().getUser(userToUpdate.getNode());
+                        roster = getRoster(userToUpdate.getNode());
                     }
                     catch (UserNotFoundException e) {
                         continue;
                     }
-                    roster = rosterCache.get(userToUpdate.getNode());
-                }
-                // Only update rosters in memory
-                if (roster != null) {
-                    roster.addSharedUser(group, addedUser);
                 }
+                // Update roster
+                roster.addSharedUser(group, addedUser);
+
                 // Check if the roster is still not in memory
                 if (addedUserRoster == null && server.isLocal(addedUser)) {
-                    addedUserRoster =
-                            rosterCache.get(addedUser.getNode());
+                    try {
+                        addedUserRoster =
+                                getRoster(addedUser.getNode());
+                    } catch (UserNotFoundException e) {
+                    }
```

Comment:
Please don't catch exceptions without doing anything with them. Even if these exceptions cannot happen, chances are that some future code change might (by accident) cause them to trigger. At the very least log an exception, but if the exception really is "impossible", apply the fail-fast strategy: rethrow it, or wrap it in a runtime exception that's thrown.